modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/resource/v1_0/test/ObjectEntryResourceTest.java

```diff
@@ -0,0 +1,246 @@
+
+		_invoke(
+			_objectDefinition.getRESTContextPath() + "?nestedFields=r_" +
+				objectRelationship.getName(),
+			Http.Method.GET);
+
+		itemsJSONArray = jsonObject.getJSONArray("items");
+
+		Assert.assertEquals(1, itemsJSONArray.length());
+
+		itemJSONObject = itemsJSONArray.getJSONObject(0);
+
+		JSONObject relatedObjectJSONObject = itemJSONObject.getJSONObject(
+			StringBundler.concat(
+				"r_", objectRelationship.getName(), "_",
+				StringUtil.replaceLast(
+					_objectDefinition.getPKObjectFieldName(), "Id", "")));
+
+		Assert.assertEquals(
+			_OBJECT_FIELD_VALUE,
+			relatedObjectJSONObject.getString(_OBJECT_FIELD_NAME));
+
+		ObjectRelationshipLocalServiceUtil.
+			deleteObjectRelationshipMappingTableValues(
+				objectRelationship.getObjectRelationshipId(),
+				_objectEntry.getPrimaryKey(),
+				relatedObjectEntry.getPrimaryKey());
+
+		ObjectRelationshipLocalServiceUtil.deleteObjectRelationship(
+			objectRelationship);
+
+		ObjectDefinitionLocalServiceUtil.deleteObjectDefinition(
+			relatedObjectDefinition);
+	}
+
+	private ObjectEntry _addObjectEntry(String objectFieldValue)
+		throws Exception {
+
+		return ObjectEntryLocalServiceUtil.addObjectEntry(
+			TestPropsValues.getUserId(), 0,
+			_objectDefinition.getObjectDefinitionId(),
+			HashMapBuilder.<String, Serializable>put(
+				_OBJECT_FIELD_NAME, objectFieldValue
+			).build(),
+			ServiceContextTestUtil.getServiceContext());
+	}
+
+	private JSONObject _invoke(String endpoint, Http.Method httpMethod)
+		throws Exception {
+
+		System.out.println("http://localhost:8080/o/" + endpoint);
+
+		Http.Options options = new Http.Options();
+
+		options.addHeader(
+			HttpHeaders.CONTENT_TYPE, ContentTypes.APPLICATION_JSON);
+		options.addHeader(
+			"Authorization",
+			"Basic " + Base64.encode("test@liferay.com:test".getBytes()));
+		options.setLocation("http://localhost:8080/o/" + endpoint);
+
+		if (httpMethod == Http.Method.PUT) {
+			options.setPut(true);
+		}
+
+		return JSONFactoryUtil.createJSONObject(HttpUtil.URLtoString(options));
+	}
+
+	private ObjectDefinition _publishObjectDefinition(
+			List<ObjectField> objectFields)
+		throws Exception {
+
+		ObjectDefinition objectDefinition =
+			_objectDefinitionLocalService.addCustomObjectDefinition(
+				TestPropsValues.getUserId(),
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
+				"A" + RandomTestUtil.randomString(), null, null,
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
+				ObjectDefinitionConstants.SCOPE_COMPANY,
+				ObjectDefinitionConstants.STORAGE_TYPE_DEFAULT, objectFields);
+
+		return _objectDefinitionLocalService.publishCustomObjectDefinition(
+			TestPropsValues.getUserId(),
+			objectDefinition.getObjectDefinitionId());
+	}
+
+	private static final String _OBJECT_FIELD_NAME =
+		"x" + RandomTestUtil.randomString();
+
+	private static final String _OBJECT_FIELD_VALUE =
+		RandomTestUtil.randomString();
+
+	@DeleteAfterTestRun
```

Comment:
I suspect if we don't delete the object definition manually, the created tables in execution time will be alive forever as I think the Rule deletes only the row in the table. Could you test it?

Reply:
I tested it and it is ok. The annotation `@DeleteAfterTestRun` calls the `deleteObjectDefinition` method that deletes also the table